lib/diyepw/diyepw/get_tmy_epw_file.py

```diff
@@ -0,0 +1,67 @@
+import tempfile as _tempfile
+import urllib.request as _request
+import re as _re
+import os as _os
+import shutil as _shutil
+from zipfile import ZipFile as _ZipFile
+from urllib.error import URLError
+
+# Buffer for the EPW catalog, which is a large HTML file that we don't want to have to download anew every time
+# a new EPW file is requested
+_catalog_html = None
+
+def get_tmy_epw_file(wmo_index:int, output_dir:str = None):
+    """
+    Given a WMO index, retrieve a TMY (typical meteorological year) EPW file for that location
+    :param wmo_index:
+    :param output_dir: Optional output directory - if not specified, the file will be saved to a temporary directory.
+        If the directory already contains a TMY EWP file matching the requested WMO Index, then a new
+        file will not be downloaded, we will just return that file's path
+    :return: The path to the AMY EPW file
+    """
+    if output_dir is None:
+        output_dir = _tempfile.mkdtemp()
+
+    # The sources we know of for TMY EPW files are http://climate.onebuilding.org and https://energyplus.net/weather;
+    # we use the climate.onebuilding.org source here because it has all of the EPW files linked
+    # from a single page, making it relatively easy and quick to find the right file.
+    catalog_url = "http://climate.onebuilding.org/WMO_Region_4_North_and_Central_America/" \
```

Comment:
website again

Reply:
Again, this is a design decision we made knowing it would potentially require ongoing support, but the assessment of the team was that these are reliably stable resources and we don't currently have an alternative. At the end of the day, going and getting these resources, then combining them together into an AMY EPW, is the capability we're offering with this tool. Downloading the data ourselves and providing it is the only alternative we came up with, and we don't have a license to do that.